### ISSUE TITLE
App screen blinks in white on iOS Safari and in black on Android Chrome

### ISSUE THREAD

**Issue by Leffe108** [flutter/flutter#86571]
## Steps to Reproduce

1. Run `flutter create bug_blink`.
2. Update the files as follows: ...

<details>
  <summary>pubspec.yaml</summary>

Add video_player and video_player_web.
```yaml
...

dependencies:
  flutter:
    sdk: flutter

  video_player: ^2.1.1
  video_player_web: ^2.0.1 # temporary until video_player depend on fixed video_player_web

...
``` 
</details>

<details>
  <summary>lib/main.dart</summary>

```dart
import 'dart:html';

import 'package:bug_blink/my_video_player.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class VideoSrc {
  String label;
  String url;

  VideoSrc({required this.label, required this.url});
}

class _MyHomePageState extends State<MyHomePage> {
  static const videos = {
    'w': 'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4',
    'z': 'https://teckensprakslexikon.su.se/movies/11/z-11946-tecken.mp4',
  };
  String? _selectedVideo;
  int _playCounter = 0;

  void _playVideo(String id) {
    setState(() {
      if (id != _selectedVideo) {
        _selectedVideo = id;
      } else {
        _playCounter++;
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Blink bug'),
      ),
      backgroundColor: Colors.lightBlue,
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            if (videos.containsKey(_selectedVideo))
              MyVideoPlayer(
                videos[_selectedVideo]!,
                playCounter: _playCounter,
              )
            else
              AspectRatio(
                  aspectRatio: 16 / 9,
                  child: Container(
                    color: Colors.black,
                  )),
            Container(
              color: Colors.black,
              child: Row(
                children: [
                  for (String id in videos.keys)
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ElevatedButton.icon(
                        style: ButtonStyle(
                            backgroundColor: MaterialStateProperty.all<Color>(
                                _selectedVideo == id
                                    ? Colors.blue
                                    : Colors.grey)),
                        onPressed: () {
                          _playVideo(id);
                        },
                        icon: Icon(Icons.play_arrow),
                        label: Text(id),
                      ),
                    ),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}
``` 
</details>

<details>
  <summary>lib/my_video_player.dart</summary>

```dart
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'on_dispose.dart';
import 'package:video_player/video_player.dart';

// This widget allow passing down a new videoUrl
// which will trigger a cross-fade into the new video.
//
// It is also possible to pass down a different playCounter
// value at some point to cause the widget to re-play
// the current video.
class MyVideoPlayer extends StatefulWidget {
  final String videoUrl;
  final Function? onVideoEnded;

  /// if playCounter changes value the video will be re-started.
  final int playCounter;

  MyVideoPlayer(this.videoUrl, {this.onVideoEnded, required this.playCounter, Key? key})
      : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  VideoPlayerController? _controller;
  List<VideoPlayerController> _oldControllers = [];
  late bool _mayShowSpinner;
  bool _oldReachedEnd = false;
  late ValueKey<String> _videoKey;

  @override
  void initState() {
    this._setUrl(this.widget.videoUrl);
    this._videoKey = ValueKey<String>(this.widget.videoUrl);
    super.initState();
  }

  void _delayMayShowSpinner() {
    this._mayShowSpinner = false;
    Future<void>.delayed(Duration(milliseconds: 500), () {
      if (!this.mounted) return;
      this.setState(() {
        this._mayShowSpinner = true;
      });
    });
  }

  @override
  void didUpdateWidget(covariant MyVideoPlayer oldWidget) {
    if (this.widget.videoUrl != oldWidget.videoUrl) {
      this._setUrl(this.widget.videoUrl);
    } else if (this.widget.playCounter != oldWidget.playCounter) {
      () async {
        await this._controller!.seekTo(Duration.zero);
        await this._controller!.play();
      }();
    }
    super.didUpdateWidget(oldWidget);
  }

  Future<VideoPlayerController?> _createController(String url) async {
    // On web, we don't use VideoCache. Cache is probably better handled
    // by the ServiceWorker on web.
    
    // bug_blink: stripped out usage of VideoCache
    if (!this.mounted) return null;
    return VideoPlayerController.network(url);
  }

  /// Wait on video to load and setup event listeners.
  /// if loading fails the video is removed from videoCache.
  /// (which is why url is a required parameter)
  Future<void> _awaitLoadVideo(
    VideoPlayerController controller,
    String url,
  ) async {
    print('Play video $url');
    try {
      print('Load video $url');
      await controller.initialize();
      if (!this.mounted) return;
      if (this._oldControllers.length == 0) {
        this.setState(() {
          controller.play(); // Autoplay
        });
      }
      print('Video loaded $url');
    } catch (e) {
      if (!this.mounted) return;
      // bug_blink: stripped out usage of VideoCache
      // bug_blink: inlined showAlert:
      await showDialog(
        context: context,
        builder: (BuildContext context) {
        return AlertDialog(
          title: Text('Oops!'),
          content: Text('Failed to load video\n\n$url'),
          actions: <Widget>[
            TextButton( // bug_blink: changed to standard button
                child: Text('Ok'),
                onPressed: () {
                  // close alert
                  Navigator.of(context).pop();
                }),
          ],
        );
        },
      );
      // TODO: Signal that video loading failed?
      return;
    }
  }

  void _addListener(VideoPlayerController controller) {
    controller.addListener(this._listener);
  }

  // video_player listener.
  // call widget callback when reaching end.
  void _listener() {
    if (!this.mounted) return;
    final reachedEnd =
        this._controller!.value.position >= this._controller!.value.duration &&
        this._controller!.value.duration.inMicroseconds > 0;
    if (reachedEnd &&
        !this._oldReachedEnd &&
        this.widget.onVideoEnded != null) {
      this.widget.onVideoEnded!();
    }
    this._oldReachedEnd = reachedEnd;
  }

  Future<VideoPlayerController?> _initController(String url) async {
    if (!this.mounted) return null;
    final controller = await this._createController(url);
    if (controller == null) return null;

    await this._awaitLoadVideo(controller, url);
    if (!this.mounted) {
      controller.dispose();
      return null;
    }
    this._addListener(controller);
    if (!this.mounted) return controller;
    print('Video aspect ratio: ${controller.value.aspectRatio}');
    return controller;
  }

  Future<void> _setUrl(String url) async {
    if (!this.mounted) return;

    // Don't show spinner directly if loading is fast
    this._delayMayShowSpinner();

    if (this._controller != null) {
      this._controller!.removeListener(this._listener);
      // Animating while playing is bad for performance => pause
      this._controller!.pause();
      this._oldControllers.add(this._controller!);
    }

    final newController = await this._initController(url);
    if (!this.mounted) {
      newController?.dispose();
      return;
    }
    setState(() {
      // Start animating in new video
      this._controller = newController;
      this._videoKey = ValueKey<String>(url);
    });
  }

  @override
  void dispose() {
    this._oldControllers.forEach((controller) {
      controller.dispose();
    });
    this._oldControllers.clear();
    this._controller?.dispose();
    this._controller = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return (this._controller != null &&
                this._controller!.value.isInitialized) ||
            (this._controller != null && this._oldControllers.length > 0)
        ? AnimatedSwitcher(
            duration: Duration(milliseconds: 350),
            child: OnDispose(
              key: this._videoKey,
              callbackData: this._controller!,
              onDispose: (controller) {
                WidgetsBinding.instance!.addPostFrameCallback((_) async {
                  if (this.mounted && this._oldControllers.contains(controller)) {
                    // Dispose old controller if mounted.
                    // If we are not mounted it will be disposed soon in
                    // dispose().
                    this._oldControllers.remove(controller);
                    controller.dispose();
                    // Play new
                    this.setState(() {
                      this._controller!.play();
                    });
                  }
                });
              },
              child: AspectRatio(
                aspectRatio: this._controller!.value.aspectRatio,
                child: VideoPlayer(this._controller!),
              ),
            ),
          )
        : AspectRatio(
            aspectRatio: 4 / 3,
            child: this._mayShowSpinner
                ? Center(child: new CircularProgressIndicator())
                : Container(),
          );
  }
}
``` 
</details>

<details>
  <summary>lib/on_dispose.dart</summary>

```dart
import 'package:flutter/widgets.dart';
import 'package:video_player/video_player.dart';

/// A widget that upon dispose calls the onDispose callback
/// with callbackData.
///
/// This widget is stateful and will use the onDispose and
/// callbackData that was pased when it was first created
/// with the key
class OnDispose extends StatefulWidget {
  final Widget? child;
  final VideoPlayerController callbackData;
  final Function(VideoPlayerController) onDispose;

  const OnDispose({
    required Key key,
    required this.onDispose,
    this.child,
    required this.callbackData,
  }) : super(key: key);

  @override
  _OnDisposeState createState() => _OnDisposeState();
}

class _OnDisposeState extends State<OnDispose> {
  late VideoPlayerController callbackData;
  late Function(VideoPlayerController) onDispose;

  @override
  void initState() {
    this.callbackData = this.widget.callbackData;
    this.onDispose = this.widget.onDispose;
    super.initState();
  }

  @override
  void dispose() {
    this.onDispose(this.callbackData);
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return this.widget.child ?? Container();
  }
}
``` 
</details>

3. `flutter run -d web-server --web-hostname 0.0.0.0 --web-port 8989 --web-renderer canvaskit` (warning: creates a public web-server on your local LAN IP) OR `flutter build --release --web-renderer canvaskit`
4. find your computer IP *OR* host the build on a server somewhere *OR* use my hosted build at https://tspquiz.se/bug_blink/
5. on your iPhone in portrait mode, open safari and type in 192.168.0.1:8989, but with the correct IP for your computer or your hosted URL. 
6. when bug_blink app loads, tap on the "▶ W" button
7. when the video has ended (or before) tap on the "▶ Z" button.

Repeat step 5-7 on Android Chrome.

**Expected results:** 
It is expected that there are some white flash-effect on the video image during cross-fade due to https://github.com/flutter/flutter/issues/80075, but not outside the video frame.

**Actual results:** 
The whole browser view flashes in white on iOS safari.

https://user-images.githubusercontent.com/4630670/125996385-9ab966e6-5030-4e55-83cc-2b317056c20b.mp4

On Android Chrome, it is not as consistent, but sometimes the whole browser briefly flashes in black. It might help to have many other tabs open that uses ram or other resources to trigger it. Or perhaps a slow phone.

https://user-images.githubusercontent.com/4630670/125996081-ba45e2bd-314f-4bf3-ae4b-1aab1e994849.MOV

<details>
  <summary>Logs</summary>

```
Analyzing bug_blink...

   info - Avoid using web-only libraries outside Flutter web plugin packages - lib\main.dart:1:1 -
          avoid_web_libraries_in_flutter
   info - Unused import: 'dart:html' - lib\main.dart:1:8 - unused_import
```

```
[✓] Flutter (Channel master, 2.4.0-5.0.pre.78, on Microsoft Windows [Version 10.0.19041.1110], locale sv-SE)
    • Flutter version 2.4.0-5.0.pre.78 at C:\Users\Leif\Programmering\Libraries\flutter-master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision aeb251e1d0 (8 hours ago), 2021-07-16 06:21:04 -0400
    • Engine revision 32e3e844cd
    • Dart version 2.14.0 (build 2.14.0-321.0.dev)

[!] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/windows#android-setup for more details.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] VS Code, 32-bit edition (version 1.56.2)
    • VS Code at C:\Program Files (x86)\Microsoft VS Code
    • Flutter extension version 3.24.0

[✓] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 91.0.4472.164
    • Edge (web)   • edge   • web-javascript • Microsoft Edge 91.0.864.64

! Doctor found issues in 1 category.
```

</details>

**Comment by TahaTesser** [comment]
Hi @Leffe108 
Thanks for filing the issue, here I assembled your classes into one class for easier reproducible of the issue

Used the following code sample

<details> 
<summary>minimal code sample</summary> 

```dart


import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:video_player/video_player.dart';

void main() { 
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class VideoSrc {
  String label;
  String url;

  VideoSrc({required this.label, required this.url});
}

class _MyHomePageState extends State<MyHomePage> {
  static const videos = {
    'w': 'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4',
    'z': 'https://teckensprakslexikon.su.se/movies/11/z-11946-tecken.mp4',
  };
  String? _selectedVideo;
  int _playCounter = 0;

  void _playVideo(String id) {
    setState(() {
      if (id != _selectedVideo) {
        _selectedVideo = id;
      } else {
        _playCounter++;
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Blink bug'),
      ),
      backgroundColor: Colors.lightBlue,
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            if (videos.containsKey(_selectedVideo))
              MyVideoPlayer(
                videos[_selectedVideo]!,
                playCounter: _playCounter,
              )
            else
              AspectRatio(
                  aspectRatio: 16 / 9,
                  child: Container(
                    color: Colors.black,
                  )),
            Container(
              color: Colors.black,
              child: Row(
                children: [
                  for (String id in videos.keys)
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ElevatedButton.icon(
                        style: ButtonStyle(
                            backgroundColor: MaterialStateProperty.all<Color>(
                                _selectedVideo == id
                                    ? Colors.blue
                                    : Colors.grey)),
                        onPressed: () {
                          _playVideo(id);
                        },
                        icon: Icon(Icons.play_arrow),
                        label: Text(id),
                      ),
                    ),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}

// This widget allow passing down a new videoUrl
// which will trigger a cross-fade into the new video.
//
// It is also possible to pass down a different playCounter
// value at some point to cause the widget to re-play
// the current video.
class MyVideoPlayer extends StatefulWidget {
  final String videoUrl;
  final Function? onVideoEnded;

  /// if playCounter changes value the video will be re-started.
  final int playCounter;

  MyVideoPlayer(this.videoUrl,
      {this.onVideoEnded, required this.playCounter, Key? key})
      : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  VideoPlayerController? _controller;
  List<VideoPlayerController> _oldControllers = [];
  late bool _mayShowSpinner;
  bool _oldReachedEnd = false;
  late ValueKey<String> _videoKey;

  @override
  void initState() {
    this._setUrl(this.widget.videoUrl);
    this._videoKey = ValueKey<String>(this.widget.videoUrl);
    super.initState();
  }

  void _delayMayShowSpinner() {
    this._mayShowSpinner = false;
    Future<void>.delayed(Duration(milliseconds: 500), () {
      if (!this.mounted) return;
      this.setState(() {
        this._mayShowSpinner = true;
      });
    });
  }

  @override
  void didUpdateWidget(covariant MyVideoPlayer oldWidget) {
    if (this.widget.videoUrl != oldWidget.videoUrl) {
      this._setUrl(this.widget.videoUrl);
    } else if (this.widget.playCounter != oldWidget.playCounter) {
      () async {
        await this._controller!.seekTo(Duration.zero);
        await this._controller!.play();
      }();
    }
    super.didUpdateWidget(oldWidget);
  }

  Future<VideoPlayerController?> _createController(String url) async {
    // On web, we don't use VideoCache. Cache is probably better handled
    // by the ServiceWorker on web.

    // bug_blink: stripped out usage of VideoCache
    if (!this.mounted) return null;
    return VideoPlayerController.network(url);
  }

  /// Wait on video to load and setup event listeners.
  /// if loading fails the video is removed from videoCache.
  /// (which is why url is a required parameter)
  Future<void> _awaitLoadVideo(
    VideoPlayerController controller,
    String url,
  ) async {
    print('Play video $url');
    try {
      print('Load video $url');
      await controller.initialize();
      if (!this.mounted) return;
      if (this._oldControllers.length == 0) {
        this.setState(() {
          controller.play(); // Autoplay
        });
      }
      print('Video loaded $url');
    } catch (e) {
      if (!this.mounted) return;
      // bug_blink: stripped out usage of VideoCache
      // bug_blink: inlined showAlert:
      await showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Oops!'),
            content: Text('Failed to load video\n\n$url'),
            actions: <Widget>[
              TextButton(
                  // bug_blink: changed to standard button
                  child: Text('Ok'),
                  onPressed: () {
                    // close alert
                    Navigator.of(context).pop();
                  }),
            ],
          );
        },
      );
      // TODO: Signal that video loading failed?
      return;
    }
  }

  void _addListener(VideoPlayerController controller) {
    controller.addListener(this._listener);
  }

  // video_player listener.
  // call widget callback when reaching end.
  void _listener() {
    if (!this.mounted) return;
    final reachedEnd =
        this._controller!.value.position >= this._controller!.value.duration &&
            this._controller!.value.duration.inMicroseconds > 0;
    if (reachedEnd &&
        !this._oldReachedEnd &&
        this.widget.onVideoEnded != null) {
      this.widget.onVideoEnded!();
    }
    this._oldReachedEnd = reachedEnd;
  }

  Future<VideoPlayerController?> _initController(String url) async {
    if (!this.mounted) return null;
    final controller = await this._createController(url);
    if (controller == null) return null;

    await this._awaitLoadVideo(controller, url);
    if (!this.mounted) {
      controller.dispose();
      return null;
    }
    this._addListener(controller);
    if (!this.mounted) return controller;
    print('Video aspect ratio: ${controller.value.aspectRatio}');
    return controller;
  }

  Future<void> _setUrl(String url) async {
    if (!this.mounted) return;

    // Don't show spinner directly if loading is fast
    this._delayMayShowSpinner();

    if (this._controller != null) {
      this._controller!.removeListener(this._listener);
      // Animating while playing is bad for performance => pause
      this._controller!.pause();
      this._oldControllers.add(this._controller!);
    }

    final newController = await this._initController(url);
    if (!this.mounted) {
      newController?.dispose();
      return;
    }
    setState(() {
      // Start animating in new video
      this._controller = newController;
      this._videoKey = ValueKey<String>(url);
    });
  }

  @override
  void dispose() {
    this._oldControllers.forEach((controller) {
      controller.dispose();
    });
    this._oldControllers.clear();
    this._controller?.dispose();
    this._controller = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return (this._controller != null &&
                this._controller!.value.isInitialized) ||
            (this._controller != null && this._oldControllers.length > 0)
        ? AnimatedSwitcher(
            duration: Duration(milliseconds: 350),
            child: OnDispose(
              key: this._videoKey,
              callbackData: this._controller!,
              onDispose: (controller) {
                WidgetsBinding.instance!.addPostFrameCallback((_) async {
                  if (this.mounted &&
                      this._oldControllers.contains(controller)) {
                    // Dispose old controller if mounted.
                    // If we are not mounted it will be disposed soon in
                    // dispose().
                    this._oldControllers.remove(controller);
                    controller.dispose();
                    // Play new
                    this.setState(() {
                      this._controller!.play();
                    });
                  }
                });
              },
              child: AspectRatio(
                aspectRatio: this._controller!.value.aspectRatio,
                child: VideoPlayer(this._controller!),
              ),
            ),
          )
        : AspectRatio(
            aspectRatio: 4 / 3,
            child: this._mayShowSpinner
                ? Center(child: new CircularProgressIndicator())
                : Container(),
          );
  }
}

/// A widget that upon dispose calls the onDispose callback
/// with callbackData.
///
/// This widget is stateful and will use the onDispose and
/// callbackData that was pased when it was first created
/// with the key
class OnDispose extends StatefulWidget {
  final Widget? child;
  final VideoPlayerController callbackData;
  final Function(VideoPlayerController) onDispose;

  const OnDispose({
    required Key key,
    required this.onDispose,
    this.child,
    required this.callbackData,
  }) : super(key: key);

  @override
  _OnDisposeState createState() => _OnDisposeState();
}

class _OnDisposeState extends State<OnDispose> {
  late VideoPlayerController callbackData;
  late Function(VideoPlayerController) onDispose;

  @override
  void initState() {
    this.callbackData = this.widget.callbackData;
    this.onDispose = this.widget.onDispose;
    super.initState();
  }

  @override
  void dispose() {
    this.onDispose(this.callbackData);
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return this.widget.child ?? Container();
  }
}

``` 

</details>

I can confirm the screen flash on both platforms,  it is not easy to reproduce, I had to try several times, indeed screen flashes 

|  `stable`  | `master` |
| :---: |:---: |
|  ❌  |   ❌  | 

Check `flutter doctor -v` outputs for each channel below

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel stable, 2.2.3, on macOS 12.0 21A5284e darwin-x64, locale en-GB)
    • Flutter version 2.2.3 at /Users/tahatesser/Code/flutter_stable
    • Framework revision f4abaa0735 (3 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Users/tahatesser/Code/openjdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios            • iOS
      14.6
    • iPhone 12 Pro (mobile) • 77C62AB9-8B71-43CF-BCAB-85B0A5B52F85 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS
      12.0 21A5284e darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google
      Chrome 91.0.4472.164

! Doctor found issues in 1 category.
```

```console
[✓] Flutter (Channel master, 2.4.0-5.0.pre.87, on macOS 12.0 21A5284e darwin-x64, locale en-GB)
    • Flutter version 2.4.0-5.0.pre.87 at /Users/tahatesser/Code/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e084e9d946 (2 days ago), 2021-07-17 12:00:21 -0700
    • Engine revision 14a9e9a50e
    • Dart version 2.14.0 (build 2.14.0-321.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/tahatesser/Code/SDK
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/tahatesser/Code/SDK
    • Java binary at: /Users/tahatesser/Code/openjdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_292-b10)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to find bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0

[✓] Connected device (4 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios            • iOS 14.6
      18F72
    • iPhone 12 Pro (mobile) • 77C62AB9-8B71-43CF-BCAB-85B0A5B52F85 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 12.0
      21A5284e darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome
      91.0.4472.164

! Doctor found issues in 1 category.
```
		
</details>

✅ : No Issue ❌: Issue reproduced

**Comment by Leffe108** [comment]
I reduced the single-file example code further (removed loading indicator, unused onVideoEnd callback, simplified load error alert)

<details>
<summary>lib/main.nut</summary>

```
import 'package:flutter/material.dart';
import 'package:flutter/foundation.dart';
import 'package:video_player/video_player.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  static const videos = {
    'w': 'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4',
    'z': 'https://teckensprakslexikon.su.se/movies/11/z-11946-tecken.mp4',
  };
  String? _selectedVideo;
  int _playCounter = 0;

  void _playVideo(String id) {
    setState(() {
      if (id != _selectedVideo) {
        _selectedVideo = id;
      } else {
        _playCounter++;
      }
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Blink bug'),
      ),
      backgroundColor: Colors.lightBlue,
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            if (videos.containsKey(_selectedVideo))
              MyVideoPlayer(
                videos[_selectedVideo]!,
                playCounter: _playCounter,
              )
            else
              AspectRatio(
                  aspectRatio: 16 / 9,
                  child: Container(
                    color: Colors.black,
                  )),
            Container(
              color: Colors.black,
              child: Row(
                children: [
                  for (String id in videos.keys)
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ElevatedButton.icon(
                        style: ButtonStyle(
                            backgroundColor: MaterialStateProperty.all<Color>(
                                _selectedVideo == id
                                    ? Colors.blue
                                    : Colors.grey)),
                        onPressed: () {
                          _playVideo(id);
                        },
                        icon: Icon(Icons.play_arrow),
                        label: Text(id),
                      ),
                    ),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}

// This widget allow passing down a new videoUrl
// which will trigger a cross-fade into the new video.
//
// It is also possible to pass down a different playCounter
// value at some point to cause the widget to re-play
// the current video.
class MyVideoPlayer extends StatefulWidget {
  final String videoUrl;

  /// if playCounter changes value the video will be re-started.
  final int playCounter;

  MyVideoPlayer(this.videoUrl, {required this.playCounter, Key? key})
      : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  VideoPlayerController? _controller;
  List<VideoPlayerController> _oldControllers = [];
  late ValueKey<String> _videoKey;

  @override
  void initState() {
    this._setUrl(this.widget.videoUrl);
    this._videoKey = ValueKey<String>(this.widget.videoUrl);
    super.initState();
  }

  @override
  void didUpdateWidget(covariant MyVideoPlayer oldWidget) {
    if (this.widget.videoUrl != oldWidget.videoUrl) {
      this._setUrl(this.widget.videoUrl);
    } else if (this.widget.playCounter != oldWidget.playCounter) {
      () async {
        await this._controller!.seekTo(Duration.zero);
        await this._controller!.play();
      }();
    }
    super.didUpdateWidget(oldWidget);
  }

  Future<VideoPlayerController?> _createController(String url) async {
    if (!this.mounted) return null;
    return VideoPlayerController.network(url);
  }

  /// Wait on video to load and setup event listeners.
  /// if loading fails the video is removed from videoCache.
  /// (which is why url is a required parameter)
  Future<void> _awaitLoadVideo(
    VideoPlayerController controller,
    String url,
  ) async {
    print('Play video $url');
    try {
      print('Load video $url');
      await controller.initialize();
      if (!this.mounted) return;
      if (this._oldControllers.length == 0) {
        this.setState(() {
          controller.play(); // Autoplay
        });
      }
      print('Video loaded $url');
    } catch (e) {
      print('Failed to load video $url');
      if (!this.mounted) return;
      await showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Failed to load video\n\n$url'),
          );
        },
      );
      return;
    }
  }

  Future<VideoPlayerController?> _initController(String url) async {
    if (!this.mounted) return null;
    final controller = await this._createController(url);
    if (controller == null) return null;

    await this._awaitLoadVideo(controller, url);
    if (!this.mounted) {
      controller.dispose();
      return null;
    }
    if (!this.mounted) return controller;
    print('Video aspect ratio: ${controller.value.aspectRatio}');
    return controller;
  }

  Future<void> _setUrl(String url) async {
    if (!this.mounted) return;

    if (this._controller != null) {
      // Animating while playing is bad for performance => pause
      this._controller!.pause();
      this._oldControllers.add(this._controller!);
    }

    final newController = await this._initController(url);
    if (!this.mounted) {
      newController?.dispose();
      return;
    }
    setState(() {
      // Start animating in new video
      this._controller = newController;
      this._videoKey = ValueKey<String>(url);
    });
  }

  @override
  void dispose() {
    this._oldControllers.forEach((controller) {
      controller.dispose();
    });
    this._oldControllers.clear();
    this._controller?.dispose();
    this._controller = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return (this._controller != null &&
                this._controller!.value.isInitialized) ||
            (this._controller != null && this._oldControllers.length > 0)
        ? AnimatedSwitcher(
            duration: Duration(milliseconds: 350),
            child: OnDispose(
              key: this._videoKey,
              callbackData: this._controller!,
              onDispose: (controller) {
                WidgetsBinding.instance!.addPostFrameCallback((_) async {
                  if (this.mounted &&
                      this._oldControllers.contains(controller)) {
                    // Dispose old controller if mounted.
                    // If we are not mounted it will be disposed soon in
                    // dispose().
                    this._oldControllers.remove(controller);
                    controller.dispose();
                    // Play new
                    this.setState(() {
                      this._controller!.play();
                    });
                  }
                });
              },
              child: AspectRatio(
                aspectRatio: this._controller!.value.aspectRatio,
                child: VideoPlayer(this._controller!),
              ),
            ),
          )
        : AspectRatio(
            aspectRatio: 4 / 3,
            child: Container(),
          );
  }
}

/// A widget that upon dispose calls the onDispose callback
/// with callbackData.
///
/// This widget is stateful and will use the onDispose and
/// callbackData that was pased when it was first created
/// with the key
class OnDispose extends StatefulWidget {
  final Widget? child;
  final VideoPlayerController callbackData;
  final Function(VideoPlayerController) onDispose;

  const OnDispose({
    required Key key,
    required this.onDispose,
    this.child,
    required this.callbackData,
  }) : super(key: key);

  @override
  _OnDisposeState createState() => _OnDisposeState();
}

class _OnDisposeState extends State<OnDispose> {
  late VideoPlayerController callbackData;
  late Function(VideoPlayerController) onDispose;

  @override
  void initState() {
    this.callbackData = this.widget.callbackData;
    this.onDispose = this.widget.onDispose;
    super.initState();
  }

  @override
  void dispose() {
    this.onDispose(this.callbackData);
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return this.widget.child ?? Container();
  }
}
``` 

</details>

I did notice that the black flash on Android occur on initial load and was able to strip away a lot of code and it still happens. I start to suspect the video_player_web plugin is causing it by itself as in this simplified code there is no addPostFrameCallback and no AnimatedSwitcher widget. Just a simple video_player.

<details>
<summary>further simplified `lib/main.dart` to only play video once</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/foundation.dart';
import 'package:video_player/video_player.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  static const videos = {
    'w': 'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4',
  };
  String? _selectedVideo;

  void _playVideo(String id) {
    setState(() {
      _selectedVideo = id;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Blink bug'),
      ),
      backgroundColor: Colors.lightBlue,
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            if (videos.containsKey(_selectedVideo))
              MyVideoPlayer(
                videos[_selectedVideo]!,
                key: Key('Video'),
              )
            else
              AspectRatio(
                  aspectRatio: 16 / 9,
                  child: Container(
                    color: Colors.black,
                  )),
            Container(
              color: Colors.black,
              child: Row(
                children: [
                  for (String id in videos.keys)
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ElevatedButton.icon(
                        style: ButtonStyle(
                            backgroundColor: MaterialStateProperty.all<Color>(
                                _selectedVideo == id
                                    ? Colors.blue
                                    : Colors.grey)),
                        onPressed: () {
                          _playVideo(id);
                        },
                        icon: Icon(Icons.play_arrow),
                        label: Text(id),
                      ),
                    ),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}

class MyVideoPlayer extends StatefulWidget {
  final String videoUrl;

  MyVideoPlayer(this.videoUrl, {Key? key})
      : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  VideoPlayerController? _controller;

  @override
  void initState() {
    this._setUrl(this.widget.videoUrl);
    super.initState();
  }

  Future<VideoPlayerController?> _createController(String url) async {
    if (!this.mounted) return null;
    return VideoPlayerController.network(url);
  }

  /// Wait on video to load and setup event listeners.
  /// if loading fails the video is removed from videoCache.
  /// (which is why url is a required parameter)
  Future<void> _awaitLoadVideo(
    VideoPlayerController controller,
    String url,
  ) async {
    print('Play video $url');
    try {
      print('Load video $url');
      await controller.initialize();
      if (!this.mounted) return;
      this.setState(() {
        controller.play(); // Autoplay
      });
      print('Video loaded $url');
    } catch (e) {
      print('Failed to load video $url');
      if (!this.mounted) return;
      await showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Failed to load video\n\n$url'),
          );
        },
      );
      return;
    }
  }

  Future<VideoPlayerController?> _initController(String url) async {
    if (!this.mounted) return null;
    final controller = await this._createController(url);
    if (controller == null) return null;

    await this._awaitLoadVideo(controller, url);
    if (!this.mounted) {
      controller.dispose();
      return null;
    }
    if (!this.mounted) return controller;
    print('Video aspect ratio: ${controller.value.aspectRatio}');
    return controller;
  }

  Future<void> _setUrl(String url) async {
    if (!this.mounted) return;

    if (this._controller != null) {
      // Pause outgoing video
      this._controller!.pause();
    }

    final newController = await this._initController(url);
    if (!this.mounted) {
      newController?.dispose();
      return;
    }
    setState(() {
      // Display video
      this._controller = newController;
    });
  }

  @override
  void dispose() {
    this._controller?.dispose();
    this._controller = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return (this._controller != null &&
            this._controller!.value.isInitialized)
        ? AspectRatio(
            aspectRatio: this._controller!.value.aspectRatio,
            child: VideoPlayer(this._controller!),
          )
        : AspectRatio(
            aspectRatio: 4 / 3,
            child: Container(),
          );
  }
}
```
</details>

To further rule out my video source and code, I tried to run the [Example](https://pub.dev/packages/video_player/example) for video_player and am able to produce full screen black flash by playing the remote video (not always but maybe 1/5 - 1/8 on a Samsung S10e)
 
<details>
<summary>lib/main.nut from video_player example</summary>

```dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// ignore_for_file: public_member_api_docs

/// An example of using the plugin, controlling lifecycle and playback of the
/// video.

import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:video_player/video_player.dart';

void main() {
  runApp(
    MaterialApp(
      home: _App(),
    ),
  );
}

class _App extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
      length: 3,
      child: Scaffold(
        key: const ValueKey<String>('home_page'),
        appBar: AppBar(
          title: const Text('Video player example'),
          actions: <Widget>[
            IconButton(
              key: const ValueKey<String>('push_tab'),
              icon: const Icon(Icons.navigation),
              onPressed: () {
                Navigator.push<_PlayerVideoAndPopPage>(
                  context,
                  MaterialPageRoute<_PlayerVideoAndPopPage>(
                    builder: (BuildContext context) => _PlayerVideoAndPopPage(),
                  ),
                );
              },
            )
          ],
          bottom: const TabBar(
            isScrollable: true,
            tabs: <Widget>[
              Tab(
                icon: Icon(Icons.cloud),
                text: "Remote",
              ),
              Tab(icon: Icon(Icons.insert_drive_file), text: "Asset"),
              Tab(icon: Icon(Icons.list), text: "List example"),
            ],
          ),
        ),
        body: TabBarView(
          children: <Widget>[
            _BumbleBeeRemoteVideo(),
            _ButterFlyAssetVideo(),
            _ButterFlyAssetVideoInList(),
          ],
        ),
      ),
    );
  }
}

class _ButterFlyAssetVideoInList extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return ListView(
      children: <Widget>[
        _ExampleCard(title: "Item a"),
        _ExampleCard(title: "Item b"),
        _ExampleCard(title: "Item c"),
        _ExampleCard(title: "Item d"),
        _ExampleCard(title: "Item e"),
        _ExampleCard(title: "Item f"),
        _ExampleCard(title: "Item g"),
        Card(
            child: Column(children: <Widget>[
          Column(
            children: <Widget>[
              const ListTile(
                leading: Icon(Icons.cake),
                title: Text("Video video"),
              ),
              Stack(
                  alignment: FractionalOffset.bottomRight +
                      const FractionalOffset(-0.1, -0.1),
                  children: <Widget>[
                    _ButterFlyAssetVideo(),
                    Image.asset('assets/flutter-mark-square-64.png'),
                  ]),
            ],
          ),
        ])),
        _ExampleCard(title: "Item h"),
        _ExampleCard(title: "Item i"),
        _ExampleCard(title: "Item j"),
        _ExampleCard(title: "Item k"),
        _ExampleCard(title: "Item l"),
      ],
    );
  }
}

/// A filler card to show the video in a list of scrolling contents.
class _ExampleCard extends StatelessWidget {
  const _ExampleCard({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Card(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          ListTile(
            leading: const Icon(Icons.airline_seat_flat_angled),
            title: Text(title),
          ),
          ButtonBar(
            children: <Widget>[
              TextButton(
                child: const Text('BUY TICKETS'),
                onPressed: () {
                  /* ... */
                },
              ),
              TextButton(
                child: const Text('SELL TICKETS'),
                onPressed: () {
                  /* ... */
                },
              ),
            ],
          ),
        ],
      ),
    );
  }
}

class _ButterFlyAssetVideo extends StatefulWidget {
  @override
  _ButterFlyAssetVideoState createState() => _ButterFlyAssetVideoState();
}

class _ButterFlyAssetVideoState extends State<_ButterFlyAssetVideo> {
  late VideoPlayerController _controller;

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.asset('assets/Butterfly-209.mp4');

    _controller.addListener(() {
      setState(() {});
    });
    _controller.setLooping(true);
    _controller.initialize().then((_) => setState(() {}));
    _controller.play();
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      child: Column(
        children: <Widget>[
          Container(
            padding: const EdgeInsets.only(top: 20.0),
          ),
          const Text('With assets mp4'),
          Container(
            padding: const EdgeInsets.all(20),
            child: AspectRatio(
              aspectRatio: _controller.value.aspectRatio,
              child: Stack(
                alignment: Alignment.bottomCenter,
                children: <Widget>[
                  VideoPlayer(_controller),
                  _ControlsOverlay(controller: _controller),
                  VideoProgressIndicator(_controller, allowScrubbing: true),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _BumbleBeeRemoteVideo extends StatefulWidget {
  @override
  _BumbleBeeRemoteVideoState createState() => _BumbleBeeRemoteVideoState();
}

class _BumbleBeeRemoteVideoState extends State<_BumbleBeeRemoteVideo> {
  late VideoPlayerController _controller;

  Future<ClosedCaptionFile> _loadCaptions() async {
    final String fileContents = await DefaultAssetBundle.of(context)
        .loadString('assets/bumble_bee_captions.srt');
    return SubRipCaptionFile(fileContents);
  }

  @override
  void initState() {
    super.initState();
    _controller = VideoPlayerController.network(
      'https://flutter.github.io/assets-for-api-docs/assets/videos/bee.mp4',
      closedCaptionFile: _loadCaptions(),
      videoPlayerOptions: VideoPlayerOptions(mixWithOthers: true),
    );

    _controller.addListener(() {
      setState(() {});
    });
    _controller.setLooping(true);
    _controller.initialize();
  }

  @override
  void dispose() {
    _controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      child: Column(
        children: <Widget>[
          Container(padding: const EdgeInsets.only(top: 20.0)),
          const Text('With remote mp4'),
          Container(
            padding: const EdgeInsets.all(20),
            child: AspectRatio(
              aspectRatio: _controller.value.aspectRatio,
              child: Stack(
                alignment: Alignment.bottomCenter,
                children: <Widget>[
                  VideoPlayer(_controller),
                  ClosedCaption(text: _controller.value.caption.text),
                  _ControlsOverlay(controller: _controller),
                  VideoProgressIndicator(_controller, allowScrubbing: true),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _ControlsOverlay extends StatelessWidget {
  const _ControlsOverlay({Key? key, required this.controller})
      : super(key: key);

  static const _examplePlaybackRates = [
    0.25,
    0.5,
    1.0,
    1.5,
    2.0,
    3.0,
    5.0,
    10.0,
  ];

  final VideoPlayerController controller;

  @override
  Widget build(BuildContext context) {
    return Stack(
      children: <Widget>[
        AnimatedSwitcher(
          duration: Duration(milliseconds: 50),
          reverseDuration: Duration(milliseconds: 200),
          child: controller.value.isPlaying
              ? SizedBox.shrink()
              : Container(
                  color: Colors.black26,
                  child: Center(
                    child: Icon(
                      Icons.play_arrow,
                      color: Colors.white,
                      size: 100.0,
                    ),
                  ),
                ),
        ),
        GestureDetector(
          onTap: () {
            controller.value.isPlaying ? controller.pause() : controller.play();
          },
        ),
        Align(
          alignment: Alignment.topRight,
          child: PopupMenuButton<double>(
            initialValue: controller.value.playbackSpeed,
            tooltip: 'Playback speed',
            onSelected: (speed) {
              controller.setPlaybackSpeed(speed);
            },
            itemBuilder: (context) {
              return [
                for (final speed in _examplePlaybackRates)
                  PopupMenuItem(
                    value: speed,
                    child: Text('${speed}x'),
                  )
              ];
            },
            child: Padding(
              padding: const EdgeInsets.symmetric(
                // Using less vertical padding as the text is also longer
                // horizontally, so it feels like it would need more spacing
                // horizontally (matching the aspect ratio of the video).
                vertical: 12,
                horizontal: 16,
              ),
              child: Text('${controller.value.playbackSpeed}x'),
            ),
          ),
        ),
      ],
    );
  }
}

class _PlayerVideoAndPopPage extends StatefulWidget {
  @override
  _PlayerVideoAndPopPageState createState() => _PlayerVideoAndPopPageState();
}

class _PlayerVideoAndPopPageState extends State<_PlayerVideoAndPopPage> {
  late VideoPlayerController _videoPlayerController;
  bool startedPlaying = false;

  @override
  void initState() {
    super.initState();

    _videoPlayerController =
        VideoPlayerController.asset('assets/Butterfly-209.mp4');
    _videoPlayerController.addListener(() {
      if (startedPlaying && !_videoPlayerController.value.isPlaying) {
        Navigator.pop(context);
      }
    });
  }

  @override
  void dispose() {
    _videoPlayerController.dispose();
    super.dispose();
  }

  Future<bool> started() async {
    await _videoPlayerController.initialize();
    await _videoPlayerController.play();
    startedPlaying = true;
    return true;
  }

  @override
  Widget build(BuildContext context) {
    return Material(
      elevation: 0,
      child: Center(
        child: FutureBuilder<bool>(
          future: started(),
          builder: (BuildContext context, AsyncSnapshot<bool> snapshot) {
            if (snapshot.data == true) {
              return AspectRatio(
                aspectRatio: _videoPlayerController.value.aspectRatio,
                child: VideoPlayer(_videoPlayerController),
              );
            } else {
              return const Text('waiting for video to load');
            }
          },
        ),
      ),
    );
  }
}
```
</details>

**Comment by Leffe108** [comment]
I further notice that if I take out controller.play(), the black screen blink still occurs. So it is likely occurring when video_player_web builds the video player widget first time.

<details>
    <summary>lib/main.dart without controller.play()</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/foundation.dart';
import 'package:video_player/video_player.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key}) : super(key: key);

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  static const videos = {
    'w': 'https://teckensprakslexikon.su.se/movies/11/w-11943-tecken.mp4',
  };
  String? _selectedVideo;

  void _playVideo(String id) {
    setState(() {
      _selectedVideo = id;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Blink bug'),
      ),
      backgroundColor: Colors.lightBlue,
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          children: <Widget>[
            if (videos.containsKey(_selectedVideo))
              MyVideoPlayer(
                videos[_selectedVideo]!,
                key: Key('Video'),
              )
            else
              AspectRatio(
                  aspectRatio: 16 / 9,
                  child: Container(
                    color: Colors.black,
                  )),
            Container(
              color: Colors.black,
              child: Row(
                children: [
                  for (String id in videos.keys)
                    Padding(
                      padding: const EdgeInsets.all(8.0),
                      child: ElevatedButton.icon(
                        style: ButtonStyle(
                            backgroundColor: MaterialStateProperty.all<Color>(
                                _selectedVideo == id
                                    ? Colors.blue
                                    : Colors.grey)),
                        onPressed: () {
                          _playVideo(id);
                        },
                        icon: Icon(Icons.play_arrow),
                        label: Text(id),
                      ),
                    ),
                ],
              ),
            )
          ],
        ),
      ),
    );
  }
}

class MyVideoPlayer extends StatefulWidget {
  final String videoUrl;

  MyVideoPlayer(this.videoUrl, {Key? key})
      : super(key: key);

  @override
  _MyVideoPlayerState createState() => _MyVideoPlayerState();
}

class _MyVideoPlayerState extends State<MyVideoPlayer> {
  VideoPlayerController? _controller;

  @override
  void initState() {
    this._setUrl(this.widget.videoUrl);
    super.initState();
  }

  Future<VideoPlayerController?> _createController(String url) async {
    if (!this.mounted) return null;
    return VideoPlayerController.network(url);
  }

  /// Wait on video to load and setup event listeners.
  /// if loading fails the video is removed from videoCache.
  /// (which is why url is a required parameter)
  Future<void> _awaitLoadVideo(
    VideoPlayerController controller,
    String url,
  ) async {
    print('Play video $url');
    try {
      print('Load video $url');
      await controller.initialize();
      if (!this.mounted) return;
      print('Video loaded $url');
    } catch (e) {
      print('Failed to load video $url');
      if (!this.mounted) return;
      await showDialog(
        context: context,
        builder: (BuildContext context) {
          return AlertDialog(
            title: Text('Failed to load video\n\n$url'),
          );
        },
      );
      return;
    }
  }

  Future<VideoPlayerController?> _initController(String url) async {
    if (!this.mounted) return null;
    final controller = await this._createController(url);
    if (controller == null) return null;

    await this._awaitLoadVideo(controller, url);
    if (!this.mounted) {
      controller.dispose();
      return null;
    }
    if (!this.mounted) return controller;
    print('Video aspect ratio: ${controller.value.aspectRatio}');
    return controller;
  }

  Future<void> _setUrl(String url) async {
    if (!this.mounted) return;

    if (this._controller != null) {
      // Pause outgoing video
      this._controller!.pause();
    }

    final newController = await this._initController(url);
    if (!this.mounted) {
      newController?.dispose();
      return;
    }
    setState(() {
      // Display video
      this._controller = newController;
    });
  }

  @override
  void dispose() {
    this._controller?.dispose();
    this._controller = null;
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return (this._controller != null &&
            this._controller!.value.isInitialized)
        ? AspectRatio(
            aspectRatio: this._controller!.value.aspectRatio,
            child: VideoPlayer(this._controller!),
          )
        : AspectRatio(
            aspectRatio: 4 / 3,
            child: Container(),
          );
  }
}
```

**Comment by aytunch** [comment]
@Leffe108 thanks for filing this issue. We are having the exact same problem especially with iOS mobile using `video-player`. The app is not usable for us atm since our app is a video feed app. So when the user is scrolling through the video `PageView` we see many flickers happening.
I also can confirm it is not caused by video controller playing but more when video controllers are initializing.
When the flickers happen, we sometimes see black and white screens **but we mostly see our native splash page**. Maybe this can hint at something?

**Comment by Leffe108** [comment]
@aytunch Thank you for joining in with your experience and adding details to this bug.

>  When the flickers happen, we sometimes see black and white screens **but we mostly see our native splash page**. Maybe this can hint at something?

On which platforms do you see the issue? To my understanding web does not have a native splash screen. (but educate me if I'm wrong) Are you saying that you see this issue also on iOS and Android native? I only see the issue on mobile web on iOS (safari) and Android (chrome). Due to this issue I purchased an apple developer account and published a native iOS version of my app and I don't see this bug there nor in the Android native build.

**Comment by aytunch** [comment]
We are using the very popular [flutter_native_splash](https://pub.dev/packages?q=flutter_native_splash) package for showing native splash pages. It supports web too.
And we see the issue on iOS mobile (CanvasKit && Chrome || Safari)

**Comment by Leffe108** [comment]
I see. Perhaps a CSS/HTML splash screen can be viewed as native to the web platform. :-)

**Comment by armandsLa** [comment]
I'm experiencing the same issue but when trying to swipe-to-dismiss(on iOS) a screen with a web view in it.

**Comment by binlebin** [comment]
I am experiencing the same issue with video_player_web 2.0.4 on iPhone iOS 15.2 on both Safari and Chrome. I also use flutter_native_splash. It would flash my splash image.

This happens when the controller initializes as well as disposes. This makes my web app that contains slides of videos not usable as well.

**Comment by Leffe108** [comment]
@armandsLa & @binlebin If you experienced this bug please give the initial post a thumbs up because apparently that is how the flutter team rank issues.

And thank you @aytunch and @binlebin for the contribution on figuring out that it is the initialization/dispose of controllers that causes the blink. That surely will help anyone who want to give this issue a try to figure out a fix. Maybe it is possible to create a further more reduced reproduction case using this information. And possible increase the likelihood for blinks to occur by rapidly init/dispose for a more accurate test case to see when the issue has been solved?

**Comment by binlebin** [comment]
I investigated further, and it is likely a problem with how Flutter embed video element into the view. I stop here:

video_player_web.dart
  @override
  Widget buildView(int textureId) {
    return HtmlElementView(viewType: 'videoPlayer-$textureId');
  }

This problem is likely related to [https://github.com/flutter/flutter/issues/39980](39980)

**Comment by doppio** [comment]
I'm seeing a similar issue. I happen to be using video_player too, but I'm not certain it's related to the flashing, at least in my case. I still see flickering after removing the VideoPlayer entirely from my screen. I also noticed that I was seeing my splash screen content. It seems like Flutter is rendering a transparent frame, since we're seeing what's behind the app.

I'm only seeing this on iOS (not Android or Mac), but I see it on every browser I've tried (Safari, Chrome, Firefox). As reported above, it's happening with both CanvasKit and the HTML renderer.

**Comment by doppio** [comment]
I actually tracked my flickering issues down to `Link` widgets from the `url_launcher` package, which renders real <a> tags on the web. Swapping them out for regular Text widgets wrapped in GestureDetectors removed the flickering. It seems like this is the same issue, somehow related to HTML element views being displayed on Flutter web on iOS.

**Comment by ditman** [comment]
I have deployed the original repro case here: https://dit-tests.web.app (even forcing 'canvaskit' build, which is not recommended for iOS devices) and I cannot see the white screen flashing in iOS or Android.

Since this issue was created a new way of embedding views was released, and multiple fixes to the way canvaskit handled html element views too; this may have gotten resolved by some of the fixes?

I'm going to close this for now, but please feel free to reopen if you can still reproduce (or even better, create a new issue with a new repro case).

(I've tried with an iPhone 11 Pro on iOS 15, and an android emulator on API 28 with Chrome 69)
